python/cudf/cudf/core/column/column.py

```diff
@@ -1217,10 +1217,14 @@ def as_column(arbitrary, nan_as_null=True, dtype=None, name=None):
             null_count = nelem - nnz
         else:
             null_count = 0
-
-        return build_column(
+        col = build_column(
             data, dtype=data.dtype, mask=mask, name=name, null_count=null_count
         )
+        # Keep a reference to `arbitrary` with the underlying
+        # RMM device array, so that the memory isn't freed out
+        # from under us
+        col.data.mem._obj = arbitrary
+        return col
```

Comment:
When `as_column` builds a new Column from a given Column, it retains the underlying `Buffer` object of the original, so that's handled. I made sure that we specifically test for this.

That said, #2755 makes `as_column(column)` a no-op.

Reply:
You're attaching this to the Numba device array as opposed to the Buffer, is that intended?

Reply:
Right sorry - I meant the device array